casper/src/main/scala/coop/rchain/casper/Casper.scala

```diff
@@ -52,7 +52,6 @@ trait Casper[F[_]] {
   def dagContains(hash: BlockHash): F[Boolean]
   def bufferContains(hash: BlockHash): F[Boolean]
   def deploy(d: Signed[DeployData]): F[Either[DeployError, DeployId]]
-  def getApprovedBlock: F[BlockMessage]
```

Comment:
I missed your conversation with Anton. Why approved block can be removed? Is ApprovedStore still necessary?

Reply:
All related data to Approved block will be removed.

Approved block a special block which carries info about signatures of validators for genesis ceremony. This is like a special type of consensus for genesis block which is not based on the stake but on number of signatures.
So all this logic will be replaced by the same finalization algorithm which is used for any other block in the same way for genesis.